apache/hive
# HIVE-25733: Add check-spelling/check-spelling for `serde/`

### What changes were proposed in this pull request?
This adds https://check-spelling.dev coverage per https://github.com/apache/hive/pull/1674#issuecomment-973098911 for the `serde` directory as #2802 has landed.

### Why are the changes needed?

The goal is to prevent new typos from being introduced into `serde`.

### Does this PR introduce _any_ user-facing change?
Yes.

* Developers who make PRs to apache/hive will see an additional workflow set run.
* If they make changes to a directory which is covered by the workflow (at the time of this PR, that's `serde`), and they introduce an unexpected word, they will receive a comment implicating the unexpected word with instructions about how to add it to `expect.txt` and the commit will gain an extra ❌ status. If the word is a typo, they'll want to correct the spelling, if not, they'll want to add it to `expect.txt` before their PR should be accepted. If the branch doesn't have any typos a ✅ status for check-spelling will appear instead. (There's 
* At the moment, the workflow runs in under a minute when it passes (under 2 minutes when it needs to make a comment about unrecognized items) -- it will get a bit longer as it covers more directories (files) in the repository (currently under 15 minutes for the full repository -- for comparison the jenkins CI takes just under 4 hours for this repository).

### How was this patch tested?
https://github.com/jsoref/hive/commit/12469eddd620dea978757830206d062e6fd72826 is the test for this commit itself.

The workflow was used to develop the fixes for #2802 and others (it's software so it will have bugs, but it's in continuous use by a number of repositories).

To test check-spelling:
1. Push this commit into your own repository.
2. Ensure that actions are enabled on your fork.
3. Create a branch where you add a typo to a file it will check. Push that to GitHub, you should receive a comment attached to the commit. (If the typo base is before the commit that adds check-spelling you won't get a comment at this point, but you could still get a comment at the next step -- it's much easier to reason through adding a commit on top of check spelling than before it.)
4. Create a PR in your fork from your typo-containing branch into your base branch (with this action). This should create an additional comment (attached to the PR instead of the commit).
5. Add another commit to the branch which added the typo (either fixing the typo or adding more) and push that to the branch. You shouldn't get a comment on the commit while you have an open PR. If covered typos are resolved, you should get a ✅ for the PR, otherwise you should get a ❌ with a comment on the PR (as when you created the PR).
